integration-cli/docker_cli_build_test.go

```diff
@@ -5444,3 +5445,513 @@ func (s *DockerTrustSuite) TestBuildContextDirIsSymlink(c *check.C) {
 		c.Fatalf("build failed with exit status %d: %s", exitStatus, out)
 	}
 }
+
+func (s *DockerSuite) TestBuildBuildTimeArg(c *check.C) {
+	imgName := "bldargtest"
+	envKey := "foo"
+	envVal := "bar"
+	buildCmd := exec.Command(dockerBinary, "build", "-t", imgName, "--build-arg",
+		fmt.Sprintf("%s=%s", envKey, envVal), "-")
+	buildCmd.Stdin = strings.NewReader(fmt.Sprintf(`FROM busybox
+		ARG %s
+		RUN echo $%s
+		CMD echo $%s`, envKey, envKey, envKey))
+
+	if out, _, err := runCommandWithOutput(buildCmd); err != nil || !strings.Contains(out, envVal) {
+		if err != nil {
+			c.Fatalf("build failed to complete: %v %v", out, err)
+		}
+		c.Fatalf("failed to access environment variable in output: '%v' expected: '%v'", out, envVal)
+	}
+
+	containerName := "bldargCont"
+	runCmd := exec.Command(dockerBinary, "run", "--name", containerName, imgName)
+	if out, _, err := runCommandWithOutput(runCmd); out != "\n" || err != nil {
+		c.Fatalf("run produced invalid output: %q, expected empty string", out)
+	}
+}
+
+func (s *DockerSuite) TestBuildBuildTimeArgCacheHit(c *check.C) {
+	imgName := "bldargtest"
+	envKey := "foo"
+	envVal := "bar"
+
+	ctx, err := fakeContext(fmt.Sprintf(`FROM busybox
```

Comment:
No need for fakeContext (and `buildImageFromContextWithArgs`) here, can just use `buildImage`, and just adjust the function to take `args ...` at the end.